org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/core/IInterpreterLoader.java

```diff
@@ -0,0 +1,82 @@
+package org.metaborg.spoofax.shell.core;
+
+import org.metaborg.core.MetaborgException;
+import org.metaborg.core.language.ILanguageImpl;
+import org.metaborg.meta.lang.dynsem.interpreter.terms.ITerm;
+import org.metaborg.meta.lang.dynsem.interpreter.terms.ITermTransformer;
+import org.metaborg.spoofax.shell.util.StrategoUtil;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+
+import com.oracle.truffle.api.vm.PolyglotEngine;
+
+/**
+ * Interface for loading a DynSem interpreter.
+ */
+public interface IInterpreterLoader {
+
+    /**
+     * Loads the DynSem interpreter for the given language implementation.
+     *
+     * @param langImpl
+     *            A language implementation with a DynSem specification.
+     * @return The DynSem entrypoint for the interpreter.
+     * @throws InterpreterLoadException
+     *             When loading results in an error.
+     */
+    PolyglotEngine loadInterpreterForLanguage(ILanguageImpl langImpl)
+        throws InterpreterLoadException;
+
+    /**
+     * Returns a {@link ITerm} corresponding to the given {@link IStrategoTerm}. Relies on
+     * {@link StrategoUtil#getSortForTerm(IStrategoTerm)} for getting the term's sort.
+     *
+     * @param term
+     *            The {@link IStrategoTerm}.
+     * @return The corresponding {@link ITerm}.
+     * @throws InterpreterLoadException
+     *             When finding the {@link ITerm} results in failure.
+     */
+    ITerm getProgramTerm(IStrategoTerm term) throws InterpreterLoadException;
+
+    /**
+     * Returns the transformer which optionally transforms a given {@link IStrategoTerm} before
+     * evaluation. By default this is the {@link ITermTransformer.IDENTITY identity transformation}.
```

Comment:
This is just the way it is generated by DynSem, I don't set this default myself.